File: A/modules/SmokeBall.py
class SmokeBall:
    def __init__(self, x=0, y=0, z=0, r=10, v=3):
        self.r = r
        self.x = x
        self.y = y
        self.z = z
        self.v = v
    def time_tick(self, dt):#时间滴答滴答 你的论文写完了吗
        self.z-=dt*self.v
    def time(self,t):#未来t秒后的位置
        S_new=SmokeBall(self.x,self.y,self.z,self.r,self.v)
        S_new.z-=t*S_new.v
        return S_new
    

File: A/rust/src/modules/smokeball.rs
// Rust version of SmokeBall.py

pub struct SmokeBall {
    pub x: f64,
    pub y: f64,
    pub z: f64,
    pub r: f64,
    pub v: f64,
}

impl SmokeBall {
    pub fn new(x: f64, y: f64, z: f64, r: f64, v: f64) -> Self {
        SmokeBall { x, y, z, r, v }
    }
    pub fn time_tick(&mut self, dt: f64) {
        self.z += self.v * dt;
    }
    pub fn time(&self, t: f64) -> SmokeBall {
        SmokeBall {
            x: self.x,
            y: self.y,
            z: self.z + self.v * t,
            r: self.r,
            v: self.v,
        }
    }
}
